### ISSUE TITLE
Installed on synology but can't make it work

### ISSUE THREAD

**Issue by Antonell333** [Jackett/Jackett#913]
i am new to synology & jackett, installed latest version (0.7.806-3)  from synoCommunity but when i try to run it , status changes to "Starting..." for few seconds and then returns to "Stopped". i have the latest mono version installed (4.6.2.7-8) & running, what else should i check? 
thanks

**Comment by Laurent1974** [comment]
Hello, same problem here (also new to Syno & rockett)
I use same Jacket version (0.7.806-3) &  mono version (4.6.2.7-8). Mono is running. But When trying to "run" Jackett, it returns to "Stopped" after some seconds. My Synology is a DS213j, running DSM 6.0.2-8451 Update 7.
Unfortunately no log.txt file found in /usr/local/jackett/var/.config/Jackett (empty folder). Is the log.txt somewhere else?
I tried to get the log also from Synology DSM GUI  (Package Center Jackett 'View Log') but also says "No Data". 
Any advice welcome

**Comment by kaso17** [comment]
the log.txt shold be in the same directory as the jackett binary (jackettconole.exe).
Try logging in via shell and running jacket manually "mono JackettConsole.exe" if the log doesn't exist

**Comment by Laurent1974** [comment]
I found JackettConsole.exe in 'NAS:/volume1/@appstore/jackett/share/jackett' folder as per below, but no log.txt
`AngleSharp.dll                       JackettConsole.exe.config           Microsoft.Owin.Host.HttpListener.dll
Autofac.dll                          JackettConsole.exe.mdb              Microsoft.Owin.Hosting.dll
Autofac.Integration.Owin.dll         Jackett.dll                         Microsoft.Owin.Host.SystemWeb.dll
Autofac.Integration.WebApi.dll       Jackett.dll.mdb                     Microsoft.Owin.StaticFiles.dll
Autofac.Integration.WebApi.Owin.dll  JackettService.exe                  MonoTorrent.dll
AutoMapper.dll                       JackettService.exe.config           Newtonsoft.Json.dll
CloudFlareUtilities.dll              JackettService.exe.mdb              NLog.dll
CommandLine.dll                      JackettTray.exe                     Owin.dll
Content                              JackettTray.exe.config              README.md
CsQuery.dll                          JackettTray.exe.mdb                 System.Net.Http.Extensions.dll
CurlSharp.dll                        JackettUpdater.exe                  System.Net.Http.Formatting.dll
CurlSharp.dll.config                 JackettUpdater.exe.config           System.Net.Http.Primitives.dll
CurlSharp.dll.mdb                    JackettUpdater.exe.mdb              System.Web.Http.dll
DateTimeRoutines.dll                 LICENSE                             System.Web.Http.Owin.dll
Definitions                          Microsoft.AspNet.Identity.Core.dll  System.Web.Http.Tracing.dll
ICSharpCode.SharpZipLib.dll          Microsoft.Owin.dll                  Upstart.config
JackettConsole.exe                   Microsoft.Owin.FileSystems.dll      YamlDotNet.dll_`
In SSH I tried the command mono JacketConsole.exe, but (sorry if dumb beginner question) I get error message as follows:
`NAS:/volume1/@appstore/jackett/share/jackett$ mono JacketConsole.exe
-sh: mono: command not found`

**Comment by kaso17** [comment]
seems like mono is not in the default PATH.
Try finding the mono binary (find / -name mono) and use the full path instead

**Comment by Antonell333** [comment]
tried:  `/volume1/@appstore/mono/mono jackettConsole.exe`
got: `Segmentation fault`

couldn't find a log.txt file afterwards. i got the same result also after typing only `/volume1/@appstore/mono/mono` so i guess the poblem is mono related

**Comment by kaso17** [comment]
Not sure if you somehow have to prepare the environment before you start mono on synology?
Maybe @Dr-Bean can help?

**Comment by Dr-Bean** [comment]
Maybe I can. See https://github.com/SynoCommunity/spksrc/issues/2298#issuecomment-271910604 please, let me know how that works for you, @Antonell333.

**Comment by Dr-Bean** [comment]
Err...and for the record, DSM6 is not yet supported by SynoCommunity, we're working on that. See https://github.com/SynoCommunity/spksrc/issues/2216#issuecomment-271852804 for details. If Jackett was working previously (i.e. installed on DSM5), and mono doesn't produce a segfault when executed, then you should be good to go. Otoh, a clean install of Jackett won't work, because it's missing the `jackett` user.

**Comment by Laurent1974** [comment]
I had the same Segmentation fault: when typing `/volume1/@appstore/mono/bin/mono`, I got `Segmentation fault`
I uninstalled Jackett and Mono, an re-installed the latest 4.6.2.7-8 Mono from beta channel as per Dr-Bean comment in [SynoCommunity/spksrc#2298 (comment)](https://github.com/SynoCommunity/spksrc/issues/2298#issuecomment-271910604)
Now I don't get `segmentation fault` anymore.
Then I re-installed Jackett Beta. Still the same problem that when i try to run Jackett, status changes to "Starting..." for few tens of seconds and then returns to "Stopped". Still no log.txt visible. So, back to original advice above, tried to generate log.txt, I could now try: 
`/volume1/@appstore/mono/bin/mono JackettConsole.exe`
Still can't find any log.txt file in JackettConsole.exe folder, but I got the following
`No fast tls on device. Using fallbacks. Current implementation :
0xee1d0f70 0xe12fff1e 0xe320f000 0xe320f000 0xe59f3018 0xe3a01000 0xe59f2014 0xe08f3003 0xe7932002 0xe3520000 0x15922000 0xeaf0e01e 0xc32f4 0x4dc 0xe1a02001 0xe1a01000
01-11 22:06:57 Error Top level exception Autofac.Core.DependencyResolutionException: An exception was thrown while invoking the constructor 'Void .ctor(ISerializeService, IProcessService, Logger)' on type 'ConfigurationService'. ---> Could not create settings directory. Cannot create /var/services/homes because a file with the same name already exists. (See inner exception for details.) ---> System.Exception: Could not create settings directory. Cannot create /var/services/homes because a file with the same name already exists.
  at Jackett.Services.ConfigurationService.CreateOrMigrateSettings () [0x0008a] in <fdfec3f4f2ea4bd0bf96b561bd9a7840>:0
  at Jackett.Services.ConfigurationService..ctor (Jackett.Services.ISerializeService s, Jackett.Services.IProcessService p, NLog.Logger l) [0x0001b] in <fdfec3f4f2ea4bd0bf96b561bd9a7840>:0
  at (wrapper dynamic-method) System.Object:lambda_method (System.Runtime.CompilerServices.Closure,object[])
  at Autofac.Core.Activators.Reflection.ConstructorParameterBinding.Instantiate () [0x00075] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
   --- End of inner exception stack trace ---
  at Autofac.Core.Activators.Reflection.ConstructorParameterBinding.Instantiate () [0x00107] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Activators.Reflection.ReflectionActivator.ActivateInstance (Autofac.IComponentContext context, System.Collections.Generic.IEnumerable`1[T] parameters) [0x000b8] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.Activate (System.Collections.Generic.IEnumerable`1[T] parameters) [0x0001a] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.<Execute>b__0 () [0x00007] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Lifetime.LifetimeScope.GetOrCreateAndShare (System.Guid id, System.Func`1[TResult] creator) [0x0002f] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.Execute () [0x00047] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.ResolveOperation.GetOrCreateInstance (Autofac.Core.ISharingLifetimeScope currentOperationScope, Autofac.Core.IComponentRegistration registration, System.Collections.Generic.IEnumerable`1[T] parameters) [0x0005e] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.ResolveComponent (Autofac.Core.IComponentRegistration registration, System.Collections.Generic.IEnumerable`1[T] parameters) [0x00000] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Activators.Reflection.AutowiringParameter+<>c__DisplayClass2.<CanSupplyValue>b__0 () [0x00011] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Activators.Reflection.ConstructorParameterBinding.Instantiate () [0x00025] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Activators.Reflection.ReflectionActivator.ActivateInstance (Autofac.IComponentContext context, System.Collections.Generic.IEnumerable`1[T] parameters) [0x000b8] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.Activate (System.Collections.Generic.IEnumerable`1[T] parameters) [0x0001a] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.<Execute>b__0 () [0x00007] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Lifetime.LifetimeScope.GetOrCreateAndShare (System.Guid id, System.Func`1[TResult] creator) [0x0002f] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.Execute () [0x00047] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.ResolveOperation.GetOrCreateInstance (Autofac.Core.ISharingLifetimeScope currentOperationScope, Autofac.Core.IComponentRegistration registration, System.Collections.Generic.IEnumerable`1[T] parameters) [0x0005e] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.ResolveComponent (Autofac.Core.IComponentRegistration registration, System.Collections.Generic.IEnumerable`1[T] parameters) [0x00000] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Activators.Reflection.AutowiringParameter+<>c__DisplayClass2.<CanSupplyValue>b__0 () [0x00011] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Activators.Reflection.ConstructorParameterBinding.Instantiate () [0x00025] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Activators.Reflection.ReflectionActivator.ActivateInstance (Autofac.IComponentContext context, System.Collections.Generic.IEnumerable`1[T] parameters) [0x000b8] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.Activate (System.Collections.Generic.IEnumerable`1[T] parameters) [0x0001a] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.<Execute>b__0 () [0x00007] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Lifetime.LifetimeScope.GetOrCreateAndShare (System.Guid id, System.Func`1[TResult] creator) [0x0002f] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.InstanceLookup.Execute () [0x00047] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.ResolveOperation.GetOrCreateInstance (Autofac.Core.ISharingLifetimeScope currentOperationScope, Autofac.Core.IComponentRegistration registration, System.Collections.Generic.IEnumerable`1[T] parameters) [0x0005e] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.ResolveOperation.ResolveComponent (Autofac.Core.IComponentRegistration registration, System.Collections.Generic.IEnumerable`1[T] parameters) [0x00000] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0
  at Autofac.Core.Resolving.ResolveOperation.Execute (Autofac.Core.IComponentRegistration registration, System.Collections.Generic.IEnumerable`1[T] parameters) [0x00000] in <ca1ca41f44e0487e9a15b2f64e38c7de>:0`

**Comment by Laurent1974** [comment]
Erratum: Tried once more to run Jackett, and this time the status did not return to "Stopped". Jackett is running. Looks good :+1:

**Comment by Antonell333** [comment]
so, @Laurent1974 , can you sum up the necessary steps?
and BTW thank you all  @kaso17 & @Dr-Bean for helping!

**Comment by Laurent1974** [comment]
@Antonell333, let me recap the steps I followed
- uninstalled jackett
- uninstalled mono 4.6.2.7-8 
- downloaded and re-installed the latest 4.6.2.7-8 Mono from beta channel as per Dr-Bean comment in [SynoCommunity/spksrc#2298 (comment)](https://github.com/SynoCommunity/spksrc/issues/2298#issuecomment-271910604)
- console: typed `/volume1/@appstore/mono/bin/mono` and I found out that the Segmentation fault disappeared.
- downloaded and re-installed jackett
- tried to run jacket from DSM: Still the same problem, status changes to "Starting..." for few tens of seconds and then returns to "Stopped". Still no log.txt visible. 
- console: I tried `/volume1/@appstore/mono/bin/mono JackettConsole.exe`. Output is a long list in the console (see above)
-  tried to run once more jackett from DSM: this time it works, jackett is "running". :+1: 
I hope above steps will solve your issue as well.
